packages/indexer/src/plugins/kv.ts

```diff
@@ -1,128 +1,137 @@
 
-type SqliteArgs = ISqlite.Config;
+type SqliteArgs = Database.Options & {
+  filename: string | Buffer | undefined;
+};
 
 export function kv<TFilter, TBlock, TRet>(args: SqliteArgs) {
   return defineIndexerPlugin<TFilter, TBlock, TRet>((indexer) => {
-    let db: Database;
+    let db: SqliteDatabase;
 
-    indexer.hooks.hook("run:before", async () => {
-      db = await open(args);
-      await KVStore.initialize(db);
+    indexer.hooks.hook("run:before", () => {
+      const { filename, ...sqliteOptions } = args;
+      db = new Database(filename, sqliteOptions);
+
+      KVStore.initialize(db);
     });
 
-    indexer.hooks.hook("handler:before", async ({ finality, endCursor }) => {
+    indexer.hooks.hook("handler:before", ({ finality, endCursor }) => {
       const ctx = useIndexerContext();
 
       assert(endCursor, new Error("endCursor cannot be undefined"));
 
       ctx.kv = new KVStore(db, finality, endCursor);
 
-      await ctx.kv.beginTransaction();
+      ctx.kv.beginTransaction();
     });
 
-    indexer.hooks.hook("handler:after", async () => {
+    indexer.hooks.hook("handler:after", () => {
       const ctx = useIndexerContext();
 
-      await ctx.kv.commitTransaction();
+      ctx.kv.commitTransaction();
 
       ctx.kv = null;
     });
 
-    indexer.hooks.hook("handler:exception", async () => {
+    indexer.hooks.hook("handler:exception", () => {
       const ctx = useIndexerContext();
 
-      await ctx.kv.rollbackTransaction();
+      ctx.kv.rollbackTransaction();
 
       ctx.kv = null;
     });
 
-    indexer.hooks.hook("run:after", async () => {
+    indexer.hooks.hook("run:after", () => {
       console.log("kv: ", useIndexerContext().kv);
     });
   });
 }
 
 export class KVStore {
   constructor(
-    private _db: Database,
+    private _db: SqliteDatabase,
     private _finality: DataFinality,
     private _endCursor: Cursor,
   ) {}
 
-  static async initialize(db: Database) {
-    await db.exec(`
+  static initialize(db: SqliteDatabase) {
+    db.pragma("journal_mode = WAL");
+
+    db.prepare(`
       CREATE TABLE IF NOT EXISTS kvs (
         from_block INTEGER NOT NULL,
         to_block INTEGER,
         k TEXT NOT NULL,
         v BLOB NOT NULL,
         PRIMARY KEY (from_block, k)
       );
-    `);
+    `).run();
   }
 
-  async beginTransaction() {
-    await this._db.exec("BEGIN TRANSACTION");
+  beginTransaction() {
+    this._db.prepare("BEGIN TRANSACTION").run();
   }
 
-  async commitTransaction() {
-    await this._db.exec("COMMIT TRANSACTION");
+  commitTransaction() {
+    this._db.prepare("COMMIT TRANSACTION").run();
   }
 
-  async rollbackTransaction() {
-    await this._db.exec("ROLLBACK TRANSACTION");
+  rollbackTransaction() {
+    this._db.prepare("ROLLBACK TRANSACTION").run();
   }
 
-  async get<T>(key: string): Promise<T> {
-    const row = await this._db.get<{ v: string }>(
-      `
+  get<T>(key: string): T {
+    const row = this._db
```

Comment:
Not sure how the sqlite api works, but shouldn't we prepare the query once in the constructor and then invoke the prepared query? This for all queries in `get`, `put`, and `del`.